### PR TITLE
C & C++ examples

### PR DESCRIPTION
**Pull request by modulovalue**
This PR translates the example in the readme into a living example in the example directory.
It also adds a C example that is essentially the same as the C++ example, but since they are considered different languages, I think it would be helpful to have separate examples for them.

### CODE REVIEW

File: wasm/example/c++_get_started/main.dart [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import 'dart:io';
+import 'package:wasm/wasm.dart';
+
+/// Start by installing the tools needed to build the Wasmer runtime:
```
Comment (liamappelbe) [inline wasm/example/c++_get_started/main.dart:4]: I think it'd make more sense to put all this in a README.md

Comment (modulovalue) [reply wasm/example/c++_get_started/main.dart:4]: Why?

Comment (liamappelbe) [reply wasm/example/c++_get_started/main.dart:4]: Readmes are where documentation about the package usually goes. Code comments are good for API documentation, but readmes are where people go when they want to know about things like what the package does, and how to set it up.

Comment (modulovalue) [reply wasm/example/c++_get_started/main.dart:4]: I've restored the package-level README.

File: wasm/example/c++_get_started/build.dart [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import 'dart:io';
```
Comment (kevmoo) [inline wasm/example/c++_get_started/build.dart:1]: maybe `cpp_` – just keep "weird" chars out of directories that may play weird

Comment (modulovalue) [reply wasm/example/c++_get_started/build.dart]: I've renamed the whole c++_get_started directory to cpp_get_started.

**Pull request closed without merging**